ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -756,9 +756,22 @@ - (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childCompone
 - (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
 {
   [_reactSubviews removeObject:(RNSScreenStackHeaderSubview *)childComponentView];
+  [self setViewToSnapshot:(RNSScreenStackHeaderSubview *)childComponentView];
   [childComponentView removeFromSuperview];
 }
 
+- (void)setViewToSnapshot:(RNSScreenStackHeaderSubview *)childComponentView
+{
+  UINavigationItem *navitem = _screenView.controller.navigationItem;
+  if (childComponentView.type == RNSScreenStackHeaderSubviewTypeLeft) {
+    navitem.leftBarButtonItem.customView = [navitem.leftBarButtonItem.customView snapshotViewAfterScreenUpdates:NO];
+  } else if (childComponentView.type == RNSScreenStackHeaderSubviewTypeCenter) {
+    navitem.titleView = [navitem.titleView snapshotViewAfterScreenUpdates:NO];
+  } else if (childComponentView.type == RNSScreenStackHeaderSubviewTypeRight) {
+    navitem.rightBarButtonItem.customView = [navitem.rightBarButtonItem.customView snapshotViewAfterScreenUpdates:NO];
+  }
```

Comment:
Do you know if search bar could also flicker, since it's also a subview of a header?

Reply:
Only custom react subviews flicker by disapearing before the screen transition's done. The searchbar does change color for a moment (like it's focused) but It's not related.

Reply:
Is it not too late to call it here? Shouldn't the `unmountChildComponentView` here be called after the subviews are already removed?

Reply:
I haven't noticed any problems with current order.

Reply:
The search bar animates nicely and does not disappear.